homework/21/instagram-api/src/main/java/PostsServlet.java

```diff
@@ -0,0 +1,148 @@
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.function.BiConsumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@WebServlet(name="PostsServlet", urlPatterns="/posts/*", loadOnStartup=1)
+public class PostsServlet extends HttpServlet {
+    @Override
+    public void init() throws ServletException {
+        getServletConfig().getServletContext().setAttribute("users", new HashMap<String, User>());
+        super.init();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String path = req.getPathInfo();
+        HashMap<String, User> users = (HashMap<String, User>) getServletConfig().getServletContext().getAttribute("users");
+        PrintWriter body = resp.getWriter();
+        if (path != null) {
+            Matcher pattern = Pattern.compile("(?<postId>0|[1-9]\\d*)/comments").matcher(path);
+            if (pattern.find()) {
+                int id = Integer.parseInt(pattern.group("postId"));
+                resp.setStatus(200);
+                body.println(Post.getPost(id));
+            } else {
+                resp.setStatus(400);
+                body.println("Id should be a number");
+            }
+        } else {
+            String username = req.getParameter("accountId");
+            if (users.containsKey(username)) {
+                if (!username.equals(getServletConfig().getServletContext().getAttribute("currentUser"))) {
+                    body.println(users.get(username).getPosts());
+                } else {
+                    resp.setStatus(409);
+                    body.println("Can only fetch other users' posts");
+                }
+            } else {
+                resp.setStatus(404);
+                body.println("No user with this username found!");
+            }
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String path = req.getPathInfo();
+        PrintWriter body = resp.getWriter();
+        HashMap<String, User> users = (HashMap<String, User>) getServletConfig().getServletContext().getAttribute("users");
+        if (path != null) {
+            Matcher pattern = Pattern.compile("(?<postId>0|[1-9]\\d*)/(?<action>comments|likes)").matcher(path);
+            System.out.println(path);
+            if (pattern.find()) {
+                int id = Integer.parseInt(pattern.group("postId"));
+                String currUser = (String) getServletConfig().getServletContext().getAttribute("currentUser");
+                if (currUser != null) {
+                    String action = pattern.group("action");
+                    Post postToChange = Post.getPost(id);
+                    if (!currUser.equals(postToChange.getProfile())) {
+                        if (!action.equals("comments")) {
```

Comment:
I think your code would be more understandable if you change this row to:

`if (action.equals("likes")) {`